libraries/AP_HAL_ChibiOS/hwdef/GEPRCF745BTHD/hwdef.dat

```diff
@@ -0,0 +1,177 @@
+define HAL_BATT_CURR_SCALE 28.5
+
+PC5 RSSI_ADC ADC1
+define BOARD_RSSI_ANA_PIN 15
+
+PC13 LED0 OUTPUT LOW GPIO(90) # LED
+define AP_NOTIFY_GPIO_LED_1_ENABLED 1
+define HAL_GPIO_A_LED_PIN 90
+define HAL_GPIO_LED_ON 0
+
+# In order to accommodate bi-directional dshot certain devices cannot be DMA enabled
+# NODMA indicates these devices, if you remove it they will still not be resolved for DMA
+
+# USART1
+PA10 USART1_RX USART1
+PA9  USART1_TX USART1
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_DJI_FPV
+
+# USART2
+# RC input defaults to UART to allow for bi-dir dshot
+PA2 USART2_TX USART2
+PA3 USART2_RX USART2
+define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_RCIN
+define DEFAULT_SERIAL2_BAUD 115
+
+# USART3 (BT)
+PB11 USART3_RX USART3 NODMA
+PB10 USART3_TX USART3 NODMA
+define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_None
+
+# UART4 (GPS)
+PA0 UART4_TX UART4 NODMA
+PA1 UART4_RX UART4 NODMA
+define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_GPS
+
+# UART6
+PC6 USART6_TX USART6 NODMA
+PC7 USART6_RX USART6 NODMA
+define DEFAULT_SERIAL6_PROTOCOL SerialProtocol_None
+
+# UART7
+PE7 UART7_RX UART7 NODMA
+PE8 UART7_TX UART7 NODMA
+define DEFAULT_SERIAL7_PROTOCOL SerialProtocol_None
+
+# UART8
+PE0 UART8_RX UART8 NODMA
+PE1 UART8_TX UART8 NODMA
+define DEFAULT_SERIAL8_PROTOCOL SerialProtocol_ESCTelemetry
+
+# Motors, bi-directional dshot capable
+PB0     TIM3_CH3 TIM3 PWM(1) GPIO(50)           # M1
+PB1     TIM3_CH4 TIM3 PWM(2) GPIO(51) BIDIR     # M2
+PB5     TIM3_CH2 TIM3 PWM(3) GPIO(52)           # M3
+PB4     TIM3_CH1 TIM3 PWM(4) GPIO(53) BIDIR     # M4
+PD12    TIM4_CH1 TIM4 PWM(5) GPIO(54)           # M5
+PD13    TIM4_CH2 TIM4 PWM(6) GPIO(55)           # M6
+PC8     TIM8_CH3 TIM8 PWM(7) GPIO(56) NODMA     # M7
+PC9     TIM8_CH4 TIM8 PWM(8) GPIO(57) NODMA     # M8
+
+# extra PWM outs
+PA8  TIM1_CH1 TIM1 PWM(9) GPIO(58) # led pin
+PD2 BUZZER OUTPUT GPIO(80) LOW
+define HAL_BUZZER_PIN 80
+define HAL_BUZZER_ON 1
+define HAL_BUZZER_OFF 0
+
+DMA_PRIORITY USART2*
+
+define HAL_STORAGE_SIZE 16384
+STORAGE_FLASH_PAGE 1
+
+# spi devices
+SPIDEV mpu6000    SPI1 DEVID1 MPU6000_CS  MODE3  1*MHZ  4*MHZ
+SPIDEV icm42688   SPI2 DEVID1 ICM42605_CS MODE3  2*MHZ  16*MHZ
+SPIDEV sdcard     SPI3 DEVID1 SDCARD_CS   MODE0  400*KHZ 25*MHZ
+SPIDEV osd        SPI4 DEVID4 MAX7456_CS  MODE0  10*MHZ 10*MHZ
+
+# IMU setup
+IMU Invensensev3 SPI:icm42688 ROTATION_ROLL_180
+IMU Invensense   SPI:mpu6000  ROTATION_YAW_90
+define HAL_DEFAULT_INS_FAST_SAMPLE 3
+
+# no built-in compass, but probe the i2c bus for all possible
+# external compass types
+define ALLOW_ARM_NO_COMPASS
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+define HAL_I2C_INTERNAL_MASK 0
+define HAL_COMPASS_AUTO_ROT_DEFAULT 2
+
+# one BARO
+BARO BMP280 I2C:0:0x76
+BARO DPS310 I2C:0:0x77
+BARO MS56XX I2C:0:0x77
+
+define HAL_OS_FATFS_IO 1
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+# setup for OSD
+define OSD_ENABLED 1
+define HAL_OSD_TYPE_DEFAULT 1
```

Comment:
```suggestion
define HAL_OSD_TYPE_DEFAULT 5
```
this sets up both the internal and DisplayPort OSD operation

Reply:
I've added OSD_TYPE2

Reply:
either way